sql/sys_vars.cc

```diff
@@ -3088,6 +3088,24 @@ static Sys_var_ulong Sys_query_prealloc_size(
        BLOCK_SIZE(1024), NO_MUTEX_GUARD, NOT_IN_BINLOG, ON_CHECK(0),
        ON_UPDATE(fix_thd_mem_root));
 
+static Sys_var_ulong Sys_opt_binlog_rows_event_max_size(
+     "binlog_row_event_max_size",
+     "The maximum size of a row-based binary log event in bytes. Rows will be "
+     "grouped into events smaller than this size if possible. "
+     "The value has to be a multiple of 256.",
+     GLOBAL_VAR(opt_binlog_rows_event_max_size), CMD_LINE(REQUIRED_ARG),
+     VALID_RANGE(256, UINT_MAX32 - (UINT_MAX32 % 256)), DEFAULT(8192),
```

Comment:
Note that this differs from the previously-implemented maximum:

- Previous maximum: `UINT_MAX32 - 1` = `0xFFFFFFFF - 1` = `0xFFFFFFFE` (not a multiple of 256 :man_shrugging:)
- New maximum: `UINT_MAX32 - (UINT_MAX32 % 256)` = `0xFFFFFFFF - (0xFFFFFFFF % 0xFF)` = `0xFFFFFF00` (now a multiple of 256)